Added SDK client generation template (#11)
pagopa/azure-pipeline-templates: templates/client-sdk-release/README.md

# Autogenerated Client SDK Release template

> Opinionated sequence of steps to generate the client SDK wiith models based on an OpenAPI spec. It does the following:

Reviewer: It's not actually a sequence of step, it's an entire stage. We should be explicit on that

1. generate models and client code based on a give OpenAPI spec, using [@pagopa/openapi-codegen-ts](https://www.npmjs.com/package/@pagopa/openapi-codegen-ts)
1. generate package.json based on main package.json values (like name and version), using [@pagopa/openapi-codegen-ts](https://www.npmjs.com/package/@pagopa/openapi-codegen-ts)
1. publish the package to NPM

> Be sure that this template is called **after** the main package.json as been updated.

Reviewer: Besides the typo:
```suggestion
Be sure that this template is called **after** the main package.json has been updated.
```

I think I understand the meaning: you are assuming this template would be executed in a context on which you are bumping the version of some other application. 
Anyway, I find this period a bit confusing for someone who's not as aware as we are of the ideas behind this template. 

Maybe we can refer `@pagopa/openapi-codegen-ts`'s documentation for implementation detail, or provide some use real-world example as use case (those are just options, don't read them as requests).

# Autogenerated Client SDK Release template

> A collection of one or more stages to generate the client SDK wiith models based on an OpenAPI spec. It does the following:

Reviewer: ```suggestion
A sequence of steps to generate and publish a SDK package based on a given OpenAPI spec. It does the following:
```

1. generate models and client code based on a give OpenAPI spec, using [@pagopa/openapi-codegen-ts](https://www.npmjs.com/package/@pagopa/openapi-codegen-ts)
1. generate package.json based on main package.json values (like name and version), using [@pagopa/openapi-codegen-ts](https://www.npmjs.com/package/@pagopa/openapi-codegen-ts)

> 1. publish the package to NPM

Reviewer: ```suggestion
1. publish the package to NPM

Be sure that this template is called **after** the main package.json has been updated.

## Usage

```yaml
resources:
  repositories:
    - repository: templates
      type: github
      name: pagopa/azure-devops-templates
      ref: refs/tags/v[n]

stages:
 - ...

>  - stage: Release
>    ...
>   
>  # Template for generating and deploying client SDk to NPM
>  - template: templates/client-sdk-release/template.yaml@templates 
>    parameters:
>      dependsOn: Release
>      openapiSpecPath: openapi/index.yaml

Reviewer: ```suggestion
 - stage: Publish_SDK
    jobs:
      - job: do_generate_and_publish
         steps: 
	     - template: templates/client-sdk-release/template.yaml@templates 
                parameters:
                openapiSpecPath: openapi/index.yaml
```

```

## Parameters

|param|description|default|required
|-|-|-|-|
|openapiSpecPath|The relative path of the OpenAPI spec from projectDir folder ||yes|

> |dependsOn|The of the stage that performs the versioning ||no|

Reviewer: I think we don't need this anymore
```suggestion
```

> |projectDir|The project directory, in case of multi-project repo |.|no!

Reviewer: Add other parameters
```suggestion
|projectDir|The project directory, in case of multi-project repo |`.`|no|
|npmRegistry|Use this to publish on a registry different than the NPM official. We also use this to publish on an ephemeral Verdaccio registry during tests |`https://registry.npmjs.org/`|no|
| artifactName |The name of the artifact that the job will publish, containing the package final bundle to publish from|`Bundle_SDK`|no|

```
